src/rpc/rawtransaction.cpp

```diff
@@ -1844,12 +1844,53 @@ static RPCHelpMan combinepsbt()
         psbtxs.push_back(psbtx);
     }
 
+    // Find if (and which) psbt has all the output blinding stuff set
+    unsigned int base_psbt_index = 0;
+    bool has_fully_blinded = false;
+    for (unsigned int i = 0; i < psbtxs.size(); ++i) {
+        const auto& psbt = psbtxs[i];
+        bool is_fully_blinded = true;
+        int unblinded_count = 0;
+        for (const auto& psbt_out : psbt.outputs) {
+            if (psbt_out.IsBlinded()) {
+                is_fully_blinded &= psbt_out.IsFullyBlinded();
+            } else {
+                unblinded_count++;
+            }
+        }
+        if (is_fully_blinded) {
+            base_psbt_index = i;
+            has_fully_blinded = true;
+            break;
+        }
+    }
+
+    // Swap the psbt we want to use as base with the 0'th psbt which is the position for the base psbt
+    if (base_psbt_index > 0) {
+        PartiallySignedTransaction psbt = psbtxs[base_psbt_index];
+        psbtxs.insert(psbtxs.begin(), psbt);
+        psbtxs.erase(psbtxs.begin() + base_psbt_index + 1);
```

Comment:
nit: surely we could swap more efficiently than this?

Reply:
Changed to use `std::swap`